buildkite/bazelci.py

```diff
@@ -580,25 +590,108 @@ def _get_buildkite_token(self):
             else "buildkite-untrusted-api-token",
         )
 
-    def _open_url(self, url):
+
+    def _open_url(self, url, params = None):
         try:
+            params_str = ""
+            if params:
+                for k, v in params:
+                    params_str = params_str + "&{}={}".format(k, v)
             return (
-                urllib.request.urlopen("{}?access_token={}".format(url, self._token))
+                urllib.request.urlopen("{}?access_token={}{}".format(url, self._token, params_str))
                 .read()
                 .decode("utf-8")
             )
         except urllib.error.HTTPError as ex:
             raise BuildkiteException("Failed to open {}: {} - {}".format(url, ex.code, ex.reason))
 
+
+    # Get build info for a pipeline with a given build number
+    # See https://buildkite.com/docs/apis/rest-api/builds#get-a-build
     def get_build_info(self, build_number):
         url = self._BUILD_STATUS_URL_TEMPLATE.format(self._org, self._pipeline, build_number)
         output = self._open_url(url)
         return json.loads(output)
 
+
+    # Get a list of build infos for a pipeline
+    # Some parameters can be set to filter the result.
+    # See https://buildkite.com/docs/apis/rest-api/builds#list-builds-for-a-pipeline
+    def get_build_info_list(self, params):
+        url = self._BUILD_STATUS_URL_TEMPLATE.format(self._org, self._pipeline, "")
+        output = self._open_url(url, params)
+        return json.loads(output)
+
+
     def get_build_log(self, job):
         return self._open_url(job["raw_log_url"])
 
 
+    @staticmethod
+    def __check_response(response, expected_status_code):
+        if response.status_code != expected_status_code:
+            print("Exit code:", response.status_code)
```

Comment:
Do you want print to stdout here? In most cases the Ci scripts writes to stderr

Reply:
Should have switched to eprint, thanks!